app/CMakeLists.txt

```diff
@@ -80,11 +80,20 @@ target_sources_ifdef(CONFIG_ZMK_BACKLIGHT app PRIVATE src/behaviors/behavior_bac
 target_sources_ifdef(CONFIG_ZMK_BLE app PRIVATE src/events/battery_state_changed.c)
 target_sources_ifdef(CONFIG_ZMK_BLE app PRIVATE src/battery.c)
 
+target_sources_ifdef(CONFIG_ZMK_ANIMATION app PRIVATE src/animation/color.c)
```

Comment:
I don't think we want all these drivers to depend on `CONFIG_ZMK_ANIMATION`. The issue this causes is that since these files implement drivers using `DT_INST_FOREACH_STATUS_OKAY`, if a driver is compiled when no node enables that animation in devicetree, you get a compiler warning.

We can resolve this with KConfigs specific to each animation driver, IE
```
config ZMK_ANIMATION_COMPOSE
    default y
    depends on DT_HAS_ZMK_ANIMATION_COMPOSE_ENABLED
config ZMK_ANIMATION_RIPPLE
    default y
    depends on DT_HAS_ZMK_ANIMATION_RIPPLE_ENABLED
```

These Kconfigs will only be turned on if a node with the given compatible is enabled (in which case the driver is used, and we would not get a compiler warning)